src/server/webkit.ts

```diff
@@ -236,36 +236,149 @@ function getMacVersion(): string {
   return cachedMacVersion;
 }
 
+class SequenceNumberMixer<V> {
+  static _lastSequenceNumber = 1;
+  private _values = new Map<number, V>();
+
+  generate(value: V): number {
+    const sequenceNumber = ++SequenceNumberMixer._lastSequenceNumber;
+    this._values.set(sequenceNumber, value);
+    return sequenceNumber;
+  }
+
+  take(sequenceNumber: number): V | undefined {
+    const value = this._values.get(sequenceNumber);
+    this._values.delete(sequenceNumber);
+    return value;
+  }
+}
+
 function wrapTransportWithWebSocket(transport: ConnectionTransport, port: number) {
   const server = new ws.Server({ port });
-  let socket: ws | undefined;
   const guid = uuidv4();
+  const idMixer = new SequenceNumberMixer<{id: number, socket: ws}>();
+  const pendingBrowserContextCreations = new Set<number>();
+  const pendingBrowserContextDeletions = new Map<number, string>();
+  const browserContextIds = new Map<string, ws>();
+  const pageProxyIds = new Map<string, ws>();
+  const sockets = new Set<ws>();
 
-  server.on('connection', (s, req) => {
-    if (req.url !== '/' + guid) {
-      s.close();
+  transport.onmessage = message => {
+    const parsedMessage = JSON.parse(message);
+    if ('id' in parsedMessage) {
+      if (parsedMessage.id === -9999)
```

Comment:
Use the kBrowserCloseMessageId. We can also remove its handling from wkConnection.ts.

Reply:
Still need it.